### ISSUE TITLE
Misleading implementation of 'dot'

### ISSUE THREAD

**Issue by szabo137** [QEDjl-project/QEDbase.jl#14]
The function 

```Julia
dot(::AbstractLorentzVector,::AbstractLorentzVector)
```
implemented [here](https://github.com/QEDjl-project/QEDbase.jl/blob/805a342ab09ad92ca3e2b992f64783c5fe6aa6c4/src/lorentz_vector.jl#L147C1-L154C4) seems misleading. 

First, there is a function `LinearAlgebra.dot` that has similar behavior but `QEDbase.dot` is not added to the methods of the former.
Second, it seems misleading, that `QEDbase.dot` redirects directly to `QEDbase.mdot`, which is the Minkowski inner product. 

In conclusion, I suggest removing the `QEDbase.dot` and sticking with `QEDbase.mdot` to underline the difference between Minkowski and Euclidian inner product. 

However, the implementation of `QEDbase.*` seems valuable, and clear and should stay. Maybe it would be more explicit to add this implementation directly to `Base.*` instead of using `import Base: *`.
